File: LeetCode-August-Challenge/UniqueMorseCodeWords.js
var uniqueMorseRepresentations = function(words) {
    let morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."];
    let mySet = new Set();
    for(let word of words){
        let tempStr = "";
        for(let char of word){
            let letters = char.charCodeAt(0) - 97;
            tempStr += morse[letters];
        }
        mySet.add(tempStr);
    }
    return mySet.size;
};
var uniqueMorseRepresentations1 = function(words) {
    let morse = {a : ".-", b: "-...", c: "-.-.",d: "-..", e: ".", f:"..-.", g:"--.", h:"....", i:"..", j:".---",k:"-.-",l:".-..",m:"--",n:"-.",o:"---",p:".--.",q:"--.-",r:".-.",s:"...",t:"-",u:"..-",v:"...-",w:".--",x:"-..-",y:"-.--",z:"--.."};
    return new Set(words.map(word => word.split('').map(letter => morse[letter]).join(''))).size;
};
let words = ["gin","zen","gig","msg"];
console.log(uniqueMorseRepresentations(words));
console.log(uniqueMorseRepresentations1(words));

File: LeetCode-August-Challenge/UniqueMorseCodeWords.java
import java.util.*;
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] morse = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Set<String> mySet = new HashSet<>();
        for(String word : words){
            StringBuilder tempStr = new StringBuilder();
            for(char ch : word.toCharArray()) tempStr.append(morse[ch - 'a']);
            mySet.add(tempStr.toString());
        }
        return mySet.size();
    }
}
public class UniqueMorseCodeWords {
    public static void main(String[] args) {
        String[] words = {"gin","zen","gig","msg"};
        Solution s = new Solution();
        System.out.println(s.uniqueMorseRepresentations(words));
    }
}
